storage_controller/src/service/safekeeper_service.rs

```diff
@@ -410,6 +422,18 @@ impl Service {
             .chain(tl.sk_set.iter())
             .collect::<HashSet<_>>();
 
+        // The timeline has no safekeepers: we need to delete it from the db manually,
+        // as no safekeeper reconciler will get to it
+        if all_sks.is_empty() {
```

Comment:
Can we have a tenant only with read-only timelines?
In this case similar logic is needed in `tenant_delete_safekeepers`

Reply:
We can't have that, because root timelines can't be read-only. Only non-root timelines, and there is always either at least one root timeline, or the tenant is empty.